src/client/mcp-client.ts

```
@@ -256,26 +292,26 @@ export class MCPClient implements IMCPClient {
             const listToolResult = await this.client.listTools({});
             logger.debug(`listTools result: ${JSON.stringify(listToolResult, null, 2)}`);
 
-            // Assume listToolResult.tools exists and iterate
+            // Populate tools
             if (listToolResult && listToolResult.tools) {
                 listToolResult.tools.forEach((tool: any) => {
                     if (!tool.description) {
-                        throw new Error(`Tool '${tool.name}' is missing a description`);
+                        logger.warn(`Tool '${tool.name}' is missing a description`);
```

i don't think we should change this - we should enforce tools have descriptions unless there's a good reason not to. this would lead to saiki silently skipping tools which is much harder to notice and debug

---

Made this change primarily because some of the public MCPs i tried did not have description but they worked with the just the toolname and inputSchema. Description is a good to have but not necessary param so I thought a warning may be apt; any better way to handle it?